server/server.go

```diff
@@ -0,0 +1,388 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+
+	ipfslite "github.com/hsanjuan/ipfs-lite"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	format "github.com/ipfs/go-ipld-format"
+	"google.golang.org/grpc"
+	pb "textile.io/grpc-ipfs-lite/ipfs-lite"
+)
+
+type ipfsLiteServer struct {
+	pb.UnimplementedIpfsLiteServer
+
+	peer *ipfslite.Peer
+}
+
+// StartServer starts the gRPC server
+func StartServer(peer *ipfslite.Peer, host string) error {
+	lis, err := net.Listen("tcp", host)
+	if err != nil {
+		return err
+	}
+
+	grpcServer := grpc.NewServer()
+	server := &ipfsLiteServer{
+		peer: peer,
+	}
+	pb.RegisterIpfsLiteServer(grpcServer, server)
+	return grpcServer.Serve(lis)
+}
+
+type addFileResult struct {
+	Node format.Node
+	Err  error
+}
+
+func addFile(ctx context.Context, peer *ipfslite.Peer, addParams *pb.AddParams, reader io.Reader, ch chan addFileResult) {
+	defer close(ch)
+	params := &ipfslite.AddParams{
+		Layout:    addParams.GetLayout(),
+		Chunker:   addParams.GetChunker(),
+		RawLeaves: addParams.GetRawLeaves(),
+		Hidden:    addParams.GetHidden(),
+		Shard:     addParams.GetShared(),
+		NoCopy:    addParams.GetNoCopy(),
+		HashFun:   addParams.GetHashFun(),
+	}
+	node, err := peer.AddFile(ctx, reader, params)
```

Comment:
It will block here until the `reader` completes, could take a while. The reader is being "driven" below by the `io.PipeWriter`.